In [1]:
import numpy as np, pandas as pd 

df = pd.read_csv("elevator_fault_monitoring_dataset.csv")

In [2]:
df

,Elevator_ID,Timestamp,Motor_Current,Torque,Brake_Status,Humidity,Vibration_X,Vibration_Y,Vibration_Z,Fault_Label
0,1,2024-01-01 08:00:00.000,10.789969,71.715047,0,66.637929,-0.098587,-0.213005,-0.041243,0
1,1,2024-01-01 08:00:00.250,13.911749,67.012111,0,38.388963,-0.283367,-0.464843,-0.217045,0
2,1,2024-01-01 08:00:00.500,14.049598,52.735928,0,45.443945,0.672220,0.732978,-0.206336,0
3,1,2024-01-01 08:00:00.750,10.205853,74.179189,1,38.756366,-0.035562,-0.810670,0.426225,0
4,1,2024-01-01 08:00:01.000,13.496271,56.449631,0,69.238637,-0.288142,0.415781,-0.282014,0
...,...,...,...,...,...,...,...,...,...,...
5183995,1,2024-01-16 07:59:58.750,11.201595,53.334129,0,38.684243,0.042123,-0.381266,-0.637350,0
5183996,1,2024-01-16 07:59:59.000,14.747344,75.756366,0,67.408743,0.427450,-0.146126,0.333217,0
5183997,1,2024-01-16 07:59:59.250,11.606207,63.771625,0,58.318129,-0.085478,-0.085246,-0.062877,0
5183998,1,2024-01-16 07:59:59.500,12.130453,59.318543,0,30.733627,0.262020,-0.042353,-0.867638,0


In [3]:
df.isnull().sum()

Elevator_ID      0
Timestamp        0
Motor_Current    0
Torque           0
Brake_Status     0
Humidity         0
Vibration_X      0
Vibration_Y      0
Vibration_Z      0
Fault_Label      0
dtype: int64

In [4]:
df.Fault_Label.value_counts()

Fault_Label
0    5019483
2      86400
1      78117
Name: count, dtype: int64

Total data samples: 5019483

Normal data samples: 86400

Fault data samples: 78117

In [ ]:
import seaborn as sns

pairplot = sns.pairplot(df, hue="Fault_Label")
fig = pairplot.figure  # ← use .figure instead of .get_figure()
fig.savefig('pairplot.png')